lib/matplotlib/figure.py

```diff
@@ -1523,6 +1524,190 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
                 .subplots(sharex=sharex, sharey=sharey, squeeze=squeeze,
                           subplot_kw=subplot_kw))
 
+    @staticmethod
+    def _normalize_grid_string(layout):
+        layout = inspect.cleandoc(layout)
+        return [list(ln) for ln in layout.strip('\n').split('\n')]
+
+    def quilt_layout(self, layout, *, subplot_kw=None, gridspec_kw=None,
+                     empty_sentinel='-'):
```

Comment:
Patchwork uses '#' as empty sentinel. Is it intentional to use a different default?

Without having read the docs first, I have misinterpreted the example
~~~
-a-
bAc
~~~
as `a` spanning all columns (which may or may not be a reasonable extension). Point is, I'd recommend a less "directional" / "semantically connecting" character as the empty sentinel. Possible choices include:

- `#` to mimic patchwork.
- ` `: space is an obvious choice but makes parsing a little harder (but AFAICS still managable)
- `.`: If we don't want to add the extra effort for space-parsing, a period looks almost empty. 

Additional remark: I wouldn't make the `empty_sentinel` configurable now. Keep it simple. I expect that most people can live with whatever default we choose and only few would spend the effort and clutter to define their own. Additionally, one can still work around if one cares a lot about the sentinel.
~~~
quilt_layout(
    """
    XaX
    bXc
    """.replace('X', '.'))
~~~
Finally, we can always add an `empty_sentinel` later if necessary.

Reply:
Good point about `-` looking like "span" instead of "blank".  I spent a while trying with whitespace as the blank, but it gets gnarly with both the de-denting we want to be able to do so you don't have to push these things against the left edge of the file and to work around editors having a "strip trailing whitespace" mode. 

I think it is simpler to leave `empty_sentinel` configurable than to document that the sentinel is dependent on the type of the input (I had written that out, but decided I didn't like it before I pushed it).  The reason the sentinel came in at all was @ImportanceOfBeingErnest request to be able to support integers (in numpy arrays).